agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -2098,28 +2073,12 @@ impl DeriveSchema for UntaggedOperator {
             UntaggedOperatorName::Locf => {
                 self.args[0].derive_schema(state)
             }
-            UntaggedOperatorName::Max => {
-                let schema = self.args[0].derive_schema(state)?;
-                let schema = match schema {
-                    Schema::AnyOf(a) => {
-                        // Unsat should be impossible, since we should never see AnyOf(empty_set)
-                        let schema = a.iter().max().unwrap_or(&Schema::Unsat);
-                        numeric_filter(schema.clone(), a)
-                    }
-                    _ => schema,
-                };
-                Ok(schema)
-            }
-            UntaggedOperatorName::Min => {
-                let schema = self.args[0].derive_schema(state)?;
-                let schema = match schema {
-                    Schema::AnyOf(a) => {
-                        // Unsat should be impossible, since we should never see AnyOf(empty_set)
-                        let schema = a.iter().min().unwrap_or(&Schema::Unsat);
-                        numeric_filter(schema.clone(), a)
-                    }
-                    _ => schema,
-                };
+            UntaggedOperatorName::Max
```

Comment:
max and min had greedy implementations...if you have an AnyOf, you can't just take the max value, because for example, $max: $foo is just going to return whatever foo is for a given row, even if its just MinKey.

There is probably something to be done here if args is an array, and a specific argument is strictly > than the rest, but that was more work than it was worth for now.